cypress/support/commands.ts

```diff
@@ -1,37 +1,29 @@
 // https://on.cypress.io/custom-commands
 import 'cypress-wait-until';
 import 'cypress-file-upload';
-import 'cypress-xpath';
-import { v4 as uuid } from 'uuid';
+
+import { expect } from 'chai';
 
 // Logins the user given a username and password
 Cypress.Commands.add('login', ({ username, password }) => {
-  cy.visit('/');
-  cy.get('[data-login-btn]').click();
-  cy.get('[data-login-username]').type(username);
-  cy.get('[data-login-password]').type(password);
-  cy.get('[data-login-submit]').click();
-  cy.waitUntil(() =>
-    cy
-      .xpath('//*[@id="__BVID__11___BV_modal_header_"]/button')
-      .should('be.visible')
-      .click(),
-  );
+  cy.request(
+    `/api/user/login?usernameOrEmail=${username}&password=${password}`,
+  ).then((response) => {
+    expect(response.status).to.eq(200);
+    cy.reload();
+    cy.get('.username').should('have.text', username);
```

Comment:
aquí sería mejor tener un selector más específico: tenemos _muchos_ elementos que potencialmente tienen esa clase.

```suggestion
    cy.get('header .username').should('have.text', username);
```